great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -535,7 +535,7 @@ def _add_fluent_datasource(
             )
 
         # We currently don't allow one to overwrite a datasource with this internal method
-        if datasource_name in self.data_sources.all():
+        if datasource_name in self._datasources:
```

Comment:
Nit - we should better differentiate `self._data_sources` and `self._datasources` - we should combine the two or perhaps make the latter `self._data_source_dict`